### PR TITLE
Reinitialize should allow to override existing config in collationGeneration

### PR DESCRIPTION
**Pull request by girazoki**
Currently the `Initialize` and `Reinitialize` messages in the collationGeneration subsystem fail if:
-  `Initialize` if there exists already another configuration and
- `Reinitialize` if another configuration does not exist

I propose to instead change the behaviour of `Reinitialize` to always set the config regardless of whether one exists or not.

### CODE REVIEW

**Comment by bkchr** [comment]
> We could also instead change the behaviour of `Reinitialize` to always set the config regardless of whether one exists or not.

IMO sounds better to me and reflects more what the name says.

**Comment by girazoki** [comment]
WIll change it to that then! thank you @bkchr

**Comment by girazoki** [comment]
Changed @bkchr

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
@girazoki can you write a prdoc?

**Comment by girazoki** [comment]
> @girazoki can you write a prdoc?

sure

File: prdoc/pr_4833.prdoc [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+doc:
+  - audience: Node Dev
+    description: |
+      The Reinitialize collationGeneration subsystem message currently fails if no other config is already set_block_length.
+      As it is difficult to query the collationGeneration subsystem to check when to call Initialize or Reinitialize, this PR
+      proposes that Reinitialize overrides the configuration regardless if there was one already set.
+
+crates:
+  - name: polkadot-node-collation-generation
+    bump: none
```
Comment (skunert) [inline prdoc/pr_4833.prdoc:12]: Should be patch as the behaviour changes.

**Review by skunert — APPROVED** [review]

File: prdoc/pr_4833.prdoc [hunk 2]
```diff
@@ -0,0 +1,12 @@
+title: "Reinitialize should allow to override existing config in collationGeneration"
+
+doc:
+  - audience: Node Dev
+    description: |
+      The Reinitialize collationGeneration subsystem message currently fails if no other config is already set_block_length.
```
Comment (sandreim) [inline prdoc/pr_4833.prdoc:6]:
```suggestion
      The Reinitialize collationGeneration subsystem message currently fails if no other config is already set.
```

**Review by sandreim — APPROVED** [review]

File: prdoc/pr_4833.prdoc [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+doc:
+  - audience: Node Dev
+    description: |
+      The Reinitialize collationGeneration subsystem message currently fails if no other config is already set.
+      As it is difficult to query the collationGeneration subsystem to check when to call Initialize or Reinitialize, this PR
+      proposes that Reinitialize overrides the configuration regardless if there was one already set.
+
+crates:
+  - name: polkadot-node-collation-generation
+    bump: none
```
Comment (bkchr) [inline prdoc/pr_4833.prdoc:12]:
```suggestion
    bump: minor
```

**Pull request merged**